migrations/Version20241018122213.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20241018122213 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Add signalement_id to history_entry';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // ajout de la colonne
+        $this->connection->executeStatement('ALTER TABLE history_entry ADD signalement_id INT DEFAULT NULL');
+
+        // Exécution directe de la requête via la connexion Doctrine
+        $this->connection->executeStatement('
+            UPDATE history_entry he
+            INNER JOIN affectation a ON he.entity_id = a.id
+            SET he.signalement_id = a.signalement_id
+            WHERE CAST(entity_name AS BINARY) = "App\\\\Entity\\\\Affectation"
+        ');
+
+        // définit le signalement_id pour toutes les entrées de type Affectation, donc l'affectation n'existe plus
+        $this->connection->executeStatement('
```

Comment:
J'ai un peu de mal a comprendre la requête mais quand j'exécute sur la base de prod il me reste des historique d'affectation sans signalement_id (si j'exécute `SELECT * FROM `history_entry` WHERE`entity_name` = 'App\\Entity\\Affectation' and signalement_id is null; `)

Je pense que les quelque cas ou la source est une commande ne sont pas gérable mais les update de statut des affectations devraient tous être complété il me semble

Reply:
Bien vu ! Je rajoute 2 requêtes pour traiter les cas rerstants. Et effectivement, il reste une dizaine de lignes venant de commande qui sont compliquées à compléter